x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/index.tsx

```diff
@@ -60,6 +60,9 @@ export class Waterfall extends Component<Props, State> {
     currentItem: null
   };
 
+  // TODO: Implement query param state management here in open/close methods, for
+  // linking into flyouts, clearing tab state within flyouts, etc?
+
```

Comment:
In that case this should not be a stateful component, but only persist the state for the open flyout in the url.